### ISSUE TITLE
ValueError while parsing data

### ISSUE THREAD

**Issue by HerrBerg** [pedrokiefer/aiostomp#13]
Since version 1.4.0 I constantly get the following error:

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/usr/lib/python3.6/asyncio/selector_events.py", line 732, in _read_ready
    self._protocol.data_received(data)
  File "/lib/python3.6/site-packages/aiostomp/aiostomp.py", line 388, in data_received 
    self._protocol.feed_data(data)
  File "/lib/python3.6/site-packages/aiostomp/protocol.py", line 43, in feed_data 
    pending_data = self._feed_data(pending_data)
  File "/lib/python3.6/site-packages/aiostomp/protocol.py", line 57, in _feed_data
    b''.join(self._pending_parts))
  File "/lib/python3.6/site-packages/aiostomp/protocol.py", line 75, in _parse_data
    command, data = data.split(b'\n', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

An examples of `data` at this point are (when I print them out): 
- `b'MESSAGE'`
- `b'ME'`
- `b'MESSAG'`
- `b'MES'`

I'm using Python 3.6.3 and connecting to an ActiveMQ Broker.

I'm facing this issue since 1.4.0.
1.3.0 works fine

Please let me know if I can do anything to help

**Comment by pedrokiefer** [comment]
@HerrBerg with 1.4.0 or 1.4.1?

On 1.4.1 we fixed a bug on handling heartbeats, which are `\n`. I remember seeing a pattern like yours while trying to find the bug. If it still happens on 1.4.1, then I'll take a look into it.

If you can print the contents of `pending_data` and `self._pending_parts` it will greatly help debugging. :)

**Comment by HerrBerg** [comment]
okay, I'm working with version 1.4.1 and am catching the Exception, that is thrown in line 75 (`_parse_data`), in the calling function `feed_data`. I had to redact some data but I retained the message length. It seems that the `data` gets truncated at the end. In the cases where the exception occurs `len(data)` is always 1345.
I subscribed to multiple topics in the case below:

data:
```b'stent:true\ntimestamp:1548945234003\n\n111.11.11.111\x00\nMESSAGE\ncontent-length:14\nexpires:0\ndestination:/topic/xxxxxxxxxxxxxxxxxxxxxxxxxl\nsubscription:1\npriority:4\nActiveMQ.MQTT.QoS:1\nmessage-id:ID\\cxxxxxx-35207-1543430467768-204\\c363\\c-1\\c1\\c463859\npersistent:true\ntimestamp:1548945234003\n\n222.222.22.222\x00\nMESSAGE\ncontent-length:12\nexpires:0\ndestination:/topic/xxxxxxxxxxxxxxxxxxxxxxxxxx\nsubscription:1\npriority:4\nActiveMQ.MQTT.QoS:1\nmessage-id:ID\\cxxxxxx-35207-1543430467768-204\\c363\\c-1\\c1\\c463860\npersistent:true\ntimestamp:1548945234005\n\n88.88.888.88\x00\nMESSAGE\ncontent-length:11\nexpires:0\ndestination:/topic/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\nsubscription:1\npriority:4\nActiveMQ.MQTT.QoS:1\nmessage-id:ID\\cxxxxxx-35207-1543430467768-204\\c362\\c-1\\c1\\c290793\npersistent:true\ntimestamp:1548945234005\n\n111.11.1.11\x00\nMESSAGE\ncontent-length:14\nexpires:0\ndestination:/topic/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\nsubscription:1\npriority:4\nActiveMQ.MQTT.QoS:1\nmessage-id:ID\\cxxxxxx-35207-1543430467768-204\\c362\\c-1\\c1\\c290794\npersistent:true\ntimestamp:1548945234005\n\n222.222.22.222\x00\nMESSAGE\ncontent-length:12\nexpires:0\ndestination:/topic/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\nsubscription:1\npriority:4\nActiveMQ.MQTT.QoS:1\nmessage-id:ID\\cxxxxxx-35207-1543430467768-204\\c362\\c-1\\c1\\c290795\npersistent:true\ntimestamp:1548945234005\n\n88.88.888.88\x00\nMESS'```

pending_data:
`b'\nMESS'`

self.pending_parts:
`[b'MESS']`

**Comment by pedrokiefer** [comment]
Thanks, I think I can work a patch with your data, not sure if I'll have time for it today though.

**Comment by pedrokiefer** [comment]
@HerrBerg sorry for the delay, can you please test #17 ?

**Comment by HerrBerg** [comment]
@pedrokiefer the fix seems to work fine.
I'm not seeing any issues on the patched version while I could see the issue occur on a 1.4.1 version running in parallel.
Great. Thanks.

**Comment by pedrokiefer** [comment]
I'll merge it and make a new release.

**Comment by pedrokiefer** [comment]
@HerrBerg just released version 1.4.2.
